### PR TITLE
Salt API change

### PR DESCRIPTION
**Pull request by vkurup**
I'm not sure when this happened, but I noticed this warning in the logs today:

```
2015-06-17 01:49:03,983 [salt.loaded.int.states.file][WARNING ][8115] Use of argument owner found, "owner" is invalid, please use "user"
```

Indeed, the latest 2 versions of their docs show `user` in the docs:
http://docs.saltstack.com/en/latest/ref/states/all/salt.states.file.html#salt.states.file.directory
http://docs.saltstack.com/en/2014.7/ref/states/all/salt.states.file.html#salt.states.file.directory

### CODE REVIEW

**Comment by dpoirier** [comment]
They've been making that change all over the place. I wonder if there are other spots that need changing?

**Comment by mlavin** [comment]
This change looks good but I agree with @dpoirier that there have been a number of these types of changes in recent versions and we should be on the look out for more.

**Comment by vkurup** [comment]
Here are the remaining instances of `owner`:

```
./conf/salt/project/db/init.sls:23:    - owner: {{ pillar['project_name'] }}_{{ pillar['environment'] }}
./conf/salt/project/queue.sls:18:    - owner: {{ pillar['project_name'] }}_{{ pillar['environment'] }}
```

The first one is for `postgres_database` and is still valid.

The second one is for `rabbitmq_vhost`, and I have no idea what to do with it since the latest docs say that all options are 'deprecated': http://docs.saltstack.com/en/latest/ref/states/all/salt.states.rabbitmq_vhost.html#salt.states.rabbitmq_vhost.present

**Comment by mlavin** [comment]
It looks like the vhost permissions where moved to the `rabbitmq_user.present` state under `perms` http://docs.saltstack.com/en/latest/ref/states/all/salt.states.rabbitmq_user.html

**Comment by dpoirier** [comment]
FYI we fixed a similar one in margarita (https://github.com/caktus/margarita/pull/38) for rabbitmq_vhost.

**Comment by mlavin** [comment]
I updated the RabbitMQ states to remove the deprecated `owner` usage.

File: conf/salt/project/queue.sls [hunk 1]
```diff
@@ -4,20 +4,24 @@ include:
...
-broker-vhost:
-  rabbitmq_vhost.present:
+broker-user:
+  rabbitmq_user.present:
     - name: {{ pillar['project_name'] }}_{{ pillar['environment'] }}
-    - owner: {{ pillar['project_name'] }}_{{ pillar['environment'] }}
+    - password: {{ pillar.get('secrets', {}).get('BROKER_PASSWORD') }}
+    - force: True
+    - perms:
+      - {{ pillar['project_name'] }}_{{ pillar['environment'] }}
```
Comment (dpoirier) [inline conf/salt/project/queue.sls]: Is this valid YAML? I'd think we'd need a colon on the end of this line, since it has a list contained under it.

Comment (mlavin) [inline conf/salt/project/queue.sls]: Good catch. That is missing a colon.

**Comment by mlavin** [comment]
Updated the Salt version and tested. Works on the first run but fails on the second do to additional bugs in Salt.

**Comment by dpoirier** [comment]
Should we mention or link here to the bugs we're hitting in Salt 2015.5.3?

**Comment by mlavin** [comment]
The current Salt bugs are referenced in #182.

**Comment by vkurup** [comment]
Just FYI, @daaray and I ran into this issue on a deploy of salt 2015.8.0rc3, and applying this PR helped us get deployed, so :+1:

**Comment by mlavin** [comment]
Thanks @vkurup!

**Pull request merged**